streams-runtimes/streams-runtime-threaded/src/test/java/org/apache/streams/local/test/processors/ObjectTypeEnforcerProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.streams.local.test.processors;
+
+import com.google.common.collect.Lists;
+import org.apache.streams.core.StreamsDatum;
+import org.apache.streams.core.StreamsProcessor;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * This processor takes in an object type and enforces that every document that
+ * comes through is of that type.
+ */
+public class ObjectTypeEnforcerProcessor implements StreamsProcessor {
```

Comment:
This might be useful even outside the scope of testing so that a stream can enforce types as they flow through the system.  Not critical right now, imo, but I see some future potential for that.